rust_src/src/eval.rs

```diff
@@ -1105,4 +1108,35 @@ pub fn funcall(args: &mut [LispObject]) -> LispObject {
     val
 }
 
+/// Pop and execute entries from the unwind-protect stack until the
+/// depth COUNT is reached. Return VALUE.
+#[no_mangle]
+pub extern "C" fn unbind_to(count: libc::ptrdiff_t, value: LispObject) -> LispObject {
+    let mut current_thread = ThreadState::current_thread();
+
```

Comment:
This whole function is unsafe basically. We should just embrace it and wrap it in a block.

Reply:
Fair enough, fixed